### PR TITLE
Use MDC instead of hardcoding "key:value" in log messages.

### PR DESCRIPTION
**Pull request by pm47**
It should make debugging easier.

Formatting is not necessarily great, by default kodein logs puts one indented MDC item per line below the message. Although it may not be a bad thing on small mobile screens.

Examples:

```
13:38:52.835 INFO  IncomingPaymentHandler - generated payment request lnbcrt1500n1p3keuxupp5unfkc90kv6rz24djsf2e5ycl6zgge22qen29dxr7d9fels7njyxsdq0w4hxjapqw3jhxaqcqpjsp5lf7yykx3prkmr8d6z5nm3hyvu2pdhzrkqe9qtdlgvadllm6h8l4q9q7sqqqqqqqqqqqqqqqqqqqsqqqqqysgqmqz9g3lpvuzjwzeerq6mr7u08ge6pjcgd60gs65fwkq23wj2qfvpm2mdjrsa02939g2039jcyred823l4c4x06w59nz9e4j99ugpe2u608jqpv3gfat
    paymentHash: e4d36c15f666862555b282559a131fd0908ca940ccd456987e69539fc3d3910d
13:38:52.837 INFO  IncomingPaymentHandler - processing htlc part expiry=CltvExpiry(400144)
    paymentHash: e4d36c15f666862555b282559a131fd0908ca940ccd456987e69539fc3d3910d
    amount: 100000 msat
    totalAmount: 150000 msat
13:38:52.839 INFO  IncomingPaymentHandler - processing pay-to-open part amount=30000 msat funding=100000 sat fees=3 sat
    paymentHash: e4d36c15f666862555b282559a131fd0908ca940ccd456987e69539fc3d3910d
    amount: 30000 msat
    totalAmount: 150000 msat
13:38:52.841 INFO  IncomingPaymentHandler - processing pay-to-open part amount=20000 msat funding=100000 sat fees=2 sat
    paymentHash: e4d36c15f666862555b282559a131fd0908ca940ccd456987e69539fc3d3910d
    amount: 20000 msat
    totalAmount: 150000 msat
13:38:52.841 WARN  IncomingPaymentHandler - amount too low for a pay-to-open: 50000 msat, min is 75000 msat
    paymentHash: e4d36c15f666862555b282559a131fd0908ca940ccd456987e69539fc3d3910d
    amount: 20000 msat
    totalAmount: 150000 msat
```

```
14:14:14.364 INFO  OutgoingPaymentHandler - sending 200000 msat to 0351042257e1bf95b260049387e80585e710b32bffb061b3cf904ced8dad1758dc
    paymentId: 92b27518-f5c7-4c17-9d93-a5a70d43616d
    paymentHash: bab218e65f2dff2318922f5276b2f54d39e0b449d1f014580cbe7b69b9e46a06
    amount: 200000 msat
    recipient: 0351042257e1bf95b260049387e80585e710b32bffb061b3cf904ced8dad1758dc
14:14:14.368 INFO  RouteCalculation - routes found: [0x0x1->205000 msat]
    paymentId: 92b27518-f5c7-4c17-9d93-a5a70d43616d
    amount: 205000 msat
14:14:14.373 INFO  OutgoingPaymentHandler - sending 205000 msat to channel 1b397a0b71e34fe36e84541b70885a5b8b846c0cc1723db7a679b89f6b3f8b43
    paymentId: 92b27518-f5c7-4c17-9d93-a5a70d43616d
    paymentHash: bab218e65f2dff2318922f5276b2f54d39e0b449d1f014580cbe7b69b9e46a06
    amount: 200000 msat
    recipient: 0351042257e1bf95b260049387e80585e710b32bffb061b3cf904ced8dad1758dc
    childPaymentId: 7dc81700-9463-41aa-ab88-9ab4d2c7ef72
14:14:14.386 INFO  OutgoingPaymentHandler - payment successfully sent (fees=5000 msat)
    childPaymentId: 7dc81700-9463-41aa-ab88-9ab4d2c7ef72
    paymentId: 92b27518-f5c7-4c17-9d93-a5a70d43616d
    paymentHash: bab218e65f2dff2318922f5276b2f54d39e0b449d1f014580cbe7b69b9e46a06
    amount: 200000 msat
    recipient: 0351042257e1bf95b260049387e80585e710b32bffb061b3cf904ced8dad1758dc
```

### CODE REVIEW

File: src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt [hunk 1]
```diff
@@ -184,11 +184,20 @@ class Peer(
...
+            StaticParams(nodeParams, remoteNodeId),
+            currentTipFlow.filterNotNull().first().first,
+            onChainFeeratesFlow.filterNotNull().first(),
+            logger = MDCLogger(
+                logger = _channelLogger,
+                staticMdc = mapOf("remoteNodeId" to remoteNodeId) + state.mdc()
+            )
+        )
+        return state.run { ctx.process(cmd) }
+    }
```
Comment (pm47) [inline src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt:200]: This is where the channel MDC is set.

File: src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt [hunk 2]
```diff
@@ -248,28 +257,28 @@ class Peer(
             _bootChannelsFlow.value = bootChannels.map { it.channelId to it }.toMap()
             // restore channels (nb: suspends on currentTipFlow & onChainFeeratesFlow)
             val channelIds = bootChannels.map {
-                logger.info { "n:$remoteNodeId restoring ${it.channelId}" }
+                logger.info { "restoring ${it.channelId}" }
                 val state = WaitForInit
-                val (state1, actions) = state.run { channelContext().process(ChannelCommand.Restore(it as ChannelState)) }
+                val (state1, actions) = state.process(ChannelCommand.Restore(it as ChannelState))
```
Comment (pm47) [inline src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt:262]: Back to a simpler `state.process`.

File: src/commonMain/kotlin/fr/acinq/lightning/utils/logger.kt [hunk 3]
```diff
@@ -1,10 +1,77 @@
...
+    }
+
+    inline fun error(ex: Throwable? = null, mdc: Map<String, Any> = emptyMap(), msgCreator: () -> String) {
+        logger.log(level = ERROR, error = ex, msgCreator = msgCreator, meta = staticMdc + mdc)
+    }
+}
+
+/**
+ * Utility functions to build MDC for various objects without polluting main classes
+ */
```
Comment (pm47) [inline src/commonMain/kotlin/fr/acinq/lightning/utils/logger.kt:40]: Not sure if grouping the `mdc()` classes here is better, it's a separate commit that I can drop.

File: src/commonMain/kotlin/fr/acinq/lightning/channel/Channel.kt [hunk 4]
```diff
@@ -140,12 +138,11 @@ data class StaticParams(val nodeParams: NodeParams, val remoteNodeId: PublicKey)
...
     val staticParams: StaticParams,
     val currentBlockHeight: Int,
-    val currentOnChainFeerates: OnChainFeerates
+    val currentOnChainFeerates: OnChainFeerates,
+    override val logger: MDCLogger
 ): LoggingContext {
     val keyManager: KeyManager get() = staticParams.nodeParams.keyManager
     val privateKey: PrivateKey get() = staticParams.nodeParams.nodePrivateKey
-
-    override val logger: Logger get() = staticParams.nodeParams.loggerFactory.newLogger(this::class)
```
Comment (pm47) [inline src/commonMain/kotlin/fr/acinq/lightning/channel/Channel.kt:148]: This fixes a bug were we were creating a new logger for each processed message.

**Review by pm47 — COMMENTED** [review]
There a quite a lot of files modified, but little significant changes. I've highlighted them below.

File: src/commonMain/kotlin/fr/acinq/lightning/utils/logger.kt [hunk 5]
```diff
@@ -1,10 +1,96 @@
...
+        state::class.simpleName?.let { put("state", it) }
+        when (state) {
+            is WaitForOpenChannel -> put("temporaryChannelId", state.temporaryChannelId)
+            is WaitForAcceptChannel -> put("temporaryChannelId", state.temporaryChannelId)
+            is WaitForFundingCreated -> put("channelId", state.channelId)
+            is WaitForFundingSigned -> put("channelId", state.channelId)
+            is ChannelStateWithCommitments -> put("channelId", state.channelId)
+            is Offline -> put("channelId", state.state.channelId)
+            is Syncing -> put("channelId", state.state.channelId)
+            else -> {}
```
Comment (t-bast) [inline src/commonMain/kotlin/fr/acinq/lightning/utils/logger.kt:82]: Shouldn't we have exhaustive pattern matching here?

**Review by t-bast — COMMENTED** [review]
Nice clean-up!

File: src/commonMain/kotlin/fr/acinq/lightning/utils/logger.kt (hunk 3 above)
Comment (t-bast) [reply src/commonMain/kotlin/fr/acinq/lightning/utils/logger.kt:48]: I think there are advantages and drawbacks to each approach and no clear winner, keeping them all here makes sense to me.

**Comment by pm47** [comment]
Rebased

File: src/commonMain/kotlin/fr/acinq/lightning/utils/logger.kt (hunk 5 above)
Comment (pm47) [reply src/commonMain/kotlin/fr/acinq/lightning/utils/logger.kt:82]: Some states like `Aborted` don't include a channel id.

Comment (t-bast) [reply src/commonMain/kotlin/fr/acinq/lightning/utils/logger.kt:82]: Yes but that's not really the point, you can have exhaustive pattern matching and not do anything for the states that don't include a `channelId`. That makes sure we're doing the right thing for all states, whereas here a state that does have a `channelId` may be incorrectly included in the `else` branch. But it's nit, it looks like we do list all the states that have a `channelId`, so it's not very important.

**Review by t-bast — APPROVED** [review]
LGTM

**Pull request merged**